microsoft/testsuites/network/sriov.py

```diff
@@ -542,6 +544,158 @@ def sriov_ethtool_offload_setting(self, environment: Environment) -> None:
+                )
+            )
+            matched_server_nic_info: NicInfo
+            for _, server_nic_info in vm_nics[server_node.name].items():
+                if (
+                    server_nic_info.ip_addr.rsplit(".", maxsplit=1)[0]
+                    == client_nic_info.ip_addr.rsplit(".", maxsplit=1)[0]
+                ):
+                    matched_server_nic_info = server_nic_info
+                    break
+            assert matched_server_nic_info
+            # 3. Start iperf3 for 10 minutes with 128 threads on client node.
+            client_iperf3.run_as_client(
+                server_ip=matched_server_nic_info.ip_addr,
+                seconds=120,
+                parallel_number=128,
+                client_ip=client_nic_info.ip_addr,
+            )
+            # 4. Get final interrupts sum per irq number on client node.
+            final_pci_interrupts_by_irqs = Interrupt.get_sum_of_interrupt_data_by_irqs(
+                client_node, client_nic_info.pci_slot
+            )
+            for init_interrupts_irq in initial_pci_interrupts_by_irqs:
+                init_irq_number = list(init_interrupts_irq)[0]
+                init_interrupts_value = init_interrupts_irq[init_irq_number]
+                for final_interrupts in final_pci_interrupts_by_irqs:
+                    final_irq_number = list(final_interrupts)[0]
+                    final_interrupts_value = final_interrupts[final_irq_number]
+                    if init_irq_number == final_irq_number:
+                        break
+                # 5. Compare interrupts changes, expected to see interrupts increased.
+                assert_that(final_interrupts_value).described_as(
+                    f"irq {init_irq_number} didn't have an increased interrupts count"
+                    " after iperf3 run!"
+                ).is_greater_than(init_interrupts_value)
+            # 6. Get final interrupts sum per cpu on client node.
+            final_pci_interrupts_by_cpus = Interrupt.get_sum_of_interrupt_data_by_cpus(
+                client_node, client_nic_info.pci_slot
+            )
+            for cpu, init_interrupts_value in initial_pci_interrupts_by_cpus.items():
+                final_interrupts_value = final_pci_interrupts_by_cpus[cpu]
+                # 7. Compare interrupts changes, expected to see interrupts increased.
+                assert_that(final_interrupts_value).described_as(
+                    f"CPU {cpu} didn't have an increased interrupts count"
+                    " after iperf3 run!"
+                ).is_greater_than(init_interrupts_value)
+
+    @TestCaseMetadata(
+        description="""
+        This case is to check whether the network connectivity is lost after running
+         iperf3 for 30 mins.
+
+        Steps,
+        1. Start iperf3 on server node.
+        2. Start iperf3 for 30 minutes on client node.
+        3. Do VF connection test.
+        """,
+        priority=3,
+        timeout=18000,
```

Comment:
Set its category to performance or stress like v2.

Reply:
add a new file named stress? and put this case there?

Reply:
It's ok to put test case in this file, because they may share the logic. The category should be performance or stress, not default functional.

Reply:
I used `self._metadata.category = "stress"`, any other way?

Reply:
It's like the `timeout`, `priority`, just write it in metadata like them.

Reply:
I didn't find category for TestCaseMetadata

Reply:
I see. It's defined on test suite level. So, you can define a new class, and set it on test suite level. The two test suites can share some methods.

Reply:
updated